lib/evaluators/evaluatorHelpers/BaseAggregateEvaluator.ts

```diff
@@ -1,23 +1,31 @@
 import type * as RDF from '@rdfjs/types';
-import type { Algebra } from 'sparqlalgebrajs';
+import { Algebra } from 'sparqlalgebrajs';
 import type { IAggregatorClass } from '../../aggregators';
 import { aggregators } from '../../aggregators';
-import type { BaseAggregator } from '../../aggregators/BaseAggregator';
+import type { SimpleAggregator } from '../../aggregators/BaseAggregator';
+import { WildcardCount } from '../../aggregators/WildcardCount';
 import type { SetFunction } from '../../util/Consts';
 import * as Err from '../../util/Errors';
 import type { ICompleteSharedContext } from './BaseExpressionEvaluator';
 
 export abstract class BaseAggregateEvaluator {
   protected expression: Algebra.AggregateExpression;
-  protected aggregator: BaseAggregator<any>;
+  protected aggregator: SimpleAggregator<any>;
   protected throwError = false;
   protected state: any;
+  protected isWildcard = false;
+  protected wildcardAggregator: WildcardCount | undefined;
 
   protected constructor(expr: Algebra.AggregateExpression,
     sharedContext: ICompleteSharedContext, throwError?: boolean) {
     this.expression = expr;
     this.aggregator = new aggregators[<SetFunction> expr.aggregator](expr, sharedContext);
     this.throwError = throwError || false;
+    this.isWildcard = expr.expression.expressionType === Algebra.expressionTypes.WILDCARD;
+    if (this.isWildcard) {
+      this.wildcardAggregator = new WildcardCount(expr, sharedContext);
+      this.put = this.__initWildcard.bind(this);
```

Comment:
Using `bind` should be avoided if possible, as it can cause obscure bugs.
Can we move these new methods into the `WildcardCount` class instead?

Reply:
I agree. Right now, the aggregate evaluators use `binds`, I tried to work in the same fassion as what is already there. Getting rid of the `bind`s completly is not an option I think. I could reduce the amound of  times we use it.

I don't think we should fix all the binds in this PR. 
If I don't use a bind, the code stands out against the existing approuch. I don't know if this is bad?

Reply:
I'll reflect back on this later. Maybe when altering the code in acordance to the next command, it will be easy to also change the bind behaviour.